### PR TITLE
[fix] torch_compile_debug don't log with 0

### LINKED ISSUE #99906: [pt2] TORCH_COMPILE_DEBUG=0 prints debug logs.
### 🐛 Describe the bug

Even if one sets `TORCH_COMPILE_DEBUG=0`, the debug logs are still printed. It would be handy to disable the log if set to 0 or false.

### Versions

master

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes https://github.com/pytorch/pytorch/issues/99906

Tested locally.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_dynamo/config.py [hunk 1]
```diff
@@ -98,7 +98,7 @@
 
 # Record and write an execution record of the current frame to a file
 # if an exception is encountered
-replay_record_enabled = bool(os.environ.get("TORCH_COMPILE_DEBUG", False))
+replay_record_enabled = os.environ.get("TORCH_COMPILE_DEBUG", "0") == "1"
```
Comment (kshitij12345) [inline torch/_dynamo/config.py:101]: Codebase usually follows this pattern

https://github.com/pytorch/pytorch/blob/0093df78df590a35deb784773aa2165884c1b7bd/torch/_dynamo/config.py#L20

File: torch/_dynamo/config.py [hunk 2]
```diff
@@ -98,7 +98,7 @@
 
 # Record and write an execution record of the current frame to a file
 # if an exception is encountered
-replay_record_enabled = bool(os.environ.get("TORCH_COMPILE_DEBUG", False))
```
Comment (kshitij12345) [inline torch/_dynamo/config.py:101]: This fails because `os.environ.get` returns a string and only empty string is `False`

**Comment by kshitij12345** [comment]
@mlazos would it make sense to read the environment variable at a single place and use that everywhere else Or is there a reason for getting/inspecting the value in different files?

**Comment by mlazos** [comment]
> @mlazos would it make sense to read the environment variable at a single place and use that everywhere else Or is there a reason for getting/inspecting the value in different files?

Feel free to do this, I think it was just read in a few places.

**Review by mlazos — APPROVED** [review]

**Comment by kshitij12345** [comment]
Will probably follow-up on that later. Thanks!

@pytorchbot merge

**Pull request closed without merging**